src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -105,6 +105,9 @@ public async Task BulkDocs()
                 if (waitForIndexesTimeout != null)
                     command.ModifiedCollections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
+                var index = command.GetMaxClusterTransactionIndex();
+                await Database.RachisLogIndexNotifications.WaitForIndexNotification(index, token.Token);
```

Comment:
@haludi if this change will be accepted, I think this is another place that you need to take of for your PR.